cpp/include/cudf/contiguous_split.hpp

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/table/table.hpp>
+#include <cudf/types.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace cudf {
+
+/**
+ * @addtogroup column_copy
+ * @{
+ * @file
+ * @brief Table APIs for contiguous_split, pack, unpack, and metadadata
+ */
+
+/**
+ * @brief Column data in a serialized format
+ *
+ * @ingroup copy_split
+ *
+ * Contains data from an array of columns in two contiguous buffers: one on host, which contains
+ * table metadata and one on device which contains the table data.
+ */
+struct packed_columns {
+  /**
+   * @brief Host-side metadata buffer used for reconstructing columns via unpack.
+   *
+   * @ingroup copy_split
+   */
+  struct metadata {
```

Comment:
I see no reason to have this struct. It just wraps around a `std::vector<uint8_t>` so can we just use `std::vector<uint8_t>` directly?

Reply:
Yeah I just moved this code, but I see the point. I'll push a commit to remove the struct in favor of a `using`, which means the rest of the code that wants to use `metadata` as a type can still do so.

Reply:
I broke the python build with this. I think all I need to do is define the type using `ctypedef`, I am working on it.

Reply:
I think you can just remove this type completely and fix the python side, such as removing `cdef struct metadata`, replacing `metadata` with just `vector[uint8_t]`.

Reply:
Should be handled now with https://github.com/rapidsai/cudf/commit/d8dd74683e6e86f4de62456ae835720286bdd074